src/content-handlers/iiif/modules/uv-shared-module/BaseExtension.ts

```diff
@@ -825,6 +815,39 @@ export class BaseExtension<T extends BaseConfig> implements IExtension {
     };
   }
 
+  getAppUri(): string {
+    const options = this.data.config!.modules.shareDialogue.options;
+
+    const host =
+      options?.embedHost ??
+      `${window.location.protocol}//${window.location.hostname}`;
+    const port = options?.embedPort ?? window.location.port;
+    const path = options?.embedPath ?? "/uv.html";
+
+    return `${host}${port ? `:${port}` : ""}${path}`;
+  }
+
+  buildEmbedScript(
+    template: string,
+    width: number,
+    height: number,
+    hashParams: URLSearchParams
+  ): string {
+    const appUri: string = this.getAppUri();
+    const title: string = this.helper.getLabel() ?? "";
+
+    const script: string = Strings.format(
+      template,
+      appUri,
+      hashParams.toString(),
```

Comment:
Is there a compelling reason to send hashParams as a separate parameter to the template, vs. concatenating it onto appUri? If we concatenate instead of separating, then we don't need to change the existing template string everywhere, and we're less likely to break existing customizations...